lib/runtime/src/ops/topk.cc

```diff
@@ -42,396 +44,190 @@ using namespace FlexFlow::Kernels::TopK;
+                   indices_ptr,
+                   input_grad.get_float_ptr(),
+                   batch_size,
+                   length,
+                   attrs.k);
 }
 
-void TopK::serialize(Legion::Serializer &sez) const {
-  sez.serialize(this->k);
-  sez.serialize(this->sorted);
+static void backward_task(Task const *task,
+                          std::vector<PhysicalRegion> const &regions,
+                          Context ctx,
+                          Runtime *runtime) {
+  TaskArgumentAccessor acc(task, regions, ctx, runtime);
+  backward_task_impl(acc);
 }
 
-Node TopK::deserialize(FFModel &ff,
-                       Legion::Deserializer &dez,
-                       ParallelTensor inputs[],
-                       int num_inputs) {
-  assert(num_inputs == 1);
-  int k;
-  bool sorted;
-  dez.deserialize(k);
-  dez.deserialize(sorted);
-  TopKParams params;
-  params.k = k;
-  params.sorted = sorted;
-  return ff.get_or_create_node<TopK>(inputs[0], params);
+CostMetrics measure_operator_cost(SimEnvFactory const &sim_factory,
+                                  TopKAttrs const &attrs,
+                                  InputParallelTensorDesc const &input,
+                                  ProfilingSettings const &settings,
+                                  MachineView const &machine_view) {
+  auto env = sim.new_environment();
+
+  ParallelTensorShape output_shape = get_output_shapes(attrs, input.shape);
+
+  SimTaskBinding init_binding;
+  init_binding.bind_arg(ATTRS, attrs);
+
+  auto init_accessor = env.get_init_accessor(TOPK_INIT_TASK_ID, init_binding);
+  DeviceSpecific<TopKPerDeviceState> per_device_state =
+      init_task_impl(init_accessor);
+
+  SimTaskBinding fwd_binding;
+  fwd_binding.bind_arg(PER_DEVICE_STATE, per_device_state);
+  fwd_binding.bind_arg(PROFILING, profiling_settings());
+  fwd_binding.bind(INPUT, input.shape);
+  fwd_binding.bind(OUTPUT, output_shape);
+  fwd_binding.bind_arg(ATTRS, attrs);
+
+  SimTaskBinding bwd_binding = infer_bwd_binding(fwd_binding);
+
+  auto fwd_accessor = env.get_fwd_accessor(TOPK_FWD_TASK_ID, fwd_binding);
+  auto bwd_accessor = env.get_bwd_accessor(TOPK_BWD_TASK_ID, bwd_binding);
+
+  float forward_time = forward_task_impl(fwd_accessor).value();
+  float backward_time = backward_task_impl(bwd_accessor).value();
+
+  float sync_time = default_estimate_sync_time(env);
+  return make_metrics(forward_time, backward_time, sync_time, env);
 }
 
-Op *TopK::materialize(FFModel &ff,
-                      ParallelTensor inputs[],
-                      int num_inputs) const {
-  TopKParams params = get_params();
-  return new TopK(ff, params, inputs[0], this->name);
+template <>
+void register_task<TOPK_INIT_TASK_ID>() {
+  OpTaskSignature init(OpTaskType::INIT);
+
+  init.add_arg_slot<SplitAttrs>(ATTRS); // Note: this may have some question
+  init.add_return_value<SplitPerDeviceState>();
+  register_task(SPLIT_INIT_TASK_ID, "Split Init", init, init_task);
```

Comment:
```suggestion
  init.add_arg_slot<TopKAttrs>(ATTRS); // Note: this may have some question
  init.add_return_value<TopKPerDeviceState>();
  register_task(TOPK_INIT_TASK_ID, "TopK Init", init, init_task);
```
Also fix this for `fwd` and `bwd`